src/rgen/Microsoft.Macios.Generator/DataModel/AttributeCodeChange.cs

```diff
@@ -0,0 +1,86 @@
+using System;
+using System.Collections.Immutable;
+
+namespace Microsoft.Macios.Generator.DataModel;
+
+/// <summary>
+/// Structure that represents a change that was made by the user on a members attribute list that has to be
+/// reflected in the generated code.
+/// </summary>
+readonly struct AttributeCodeChange : IEquatable<AttributeCodeChange> {
+
+	/// <summary>
+	/// Get the name of the attribute that was added.
+	/// </summary>
+	public string Name { get; }
+
+	/// <summary>
+	/// Get the arguments used to create the attribute.
+	/// </summary>
+	public ImmutableArray<string> Arguments { get; }
+
+	/// <summary>
+	/// Create a new change that happened on an attribute.
+	/// </summary>
+	/// <param name="name">The name of the attribute that was added.</param>
+	/// <param name="arguments">The arguments used to create the attribute.</param>
+	public AttributeCodeChange (string name, ImmutableArray<string> arguments)
+	{
+		Name = name;
+		Arguments = arguments;
+	}
+
+	/// <summary>
+	/// Create a new change that happened on an attribute with no parameters.
+	/// </summary>
+	/// <param name="name">The name of the attribute that was added.</param>
+	public AttributeCodeChange (string name) : this (name, []) { }
+
+	/// <inheritdoc />
+	public bool Equals (AttributeCodeChange other)
+	{
+		if (Name != other.Name || Arguments.Length != other.Arguments.Length)
```

Comment:
That is an incredibly good question. The attribute name will have the fully qualified data because we are using the SemanticModel to get it, that is we are converting the syntax to the actual symbol information. But you did catch a bug since enumerators are not LiteralExpressionSyntax but MemberAccessExpressionSyntax, I am missing that symbol information.

Attributes are special and just allow the following expressions:
* LiteralExpression => string, int, byte, etc
* MemberAccessExpression => enums
* TypeOfExpressions => typeof (T)

I have improved the code to get the symbol from the MemberAccessExpression and the TypeOfExpression so that diff syntax has the same string thanks to the semantic model: https://github.com/xamarin/xamarin-macios/pull/21517/commits/926d1e6c367986b0a05cfccb0b6c49c12366db21